byzcoin/InstanceVersioning.md

```diff
@@ -0,0 +1,52 @@
+Navigation: [DEDIS](https://github.com/dedis/doc/tree/master/README.md) ::
+[Cothority](https://github.com/dedis/cothority/tree/master/README.md) ::
+[Building Blocks](https://github.com/dedis/cothority/tree/master/doc/BuildingBlocks.md) ::
+[ByzCoin](README.md) ::
+Instance Versioning
+
+# Instance Versioning
+
+This document gives a detailed overview of the instance versioning that allows
+security check and history navigation.
+
+## Versioning
+
+Each state change has a `Version` field populated with the version of the instance
+after having applied the change. It is monotonically increasing by one every time
+an instruction is performed on an instance.
+The global state also stores the version at the moment of the block creation because
+we need to know what is the latest version to create new transaction asking the right
+version. This prevents replay attacks (see #1442).
+The Byzcoin service takes care of the versioning by modifying the results of a
```

Comment:
If the byzcoin service takes care of the versioning, then you don't need to put the version in the `StateChange`. `StateChange` is created by the contract.

Reply:
I am fine with storing the version alongside the state change and the block index.